hathor/conf/settings.py

```diff
@@ -384,6 +386,14 @@ def MAXIMUM_NUMBER_OF_HALVINGS(self) -> int:
     # Time to update the peers that are running sync.
     SYNC_UPDATE_INTERVAL: int = 10 * 60  # seconds
 
+    # List of criteria for all features that participate in the feature activation process for this network,
+    # past or future, activated or not. Features should NOT be removed from this list, to preserve history.
+    FEATURE_ACTIVATION: List[Criteria] = []
+
+    # Specifies the default minimum number of blocks per evaluation interval required to activate a feature.
+    # Usually calculated from a percentage of feature_activation.constants.EVALUATION_INTERVAL.
+    DEFAULT_FEATURE_ACTIVATION_THRESHOLD: int = 36288  # 36288 = 90% of 40320
```

Comment:
Shouldn't we let this parameter be part of the criteria?

Reply:
It is! The threshold is configurable per-feature in the Criteria class: `threshold: Optional[int]`

However, it's optional, and we do provide a default value for it. That settings attribute is the default value. Since the default value is different per network, I cannot assign it at class definition, but rather get it in runtime. If `criteria.threshold` is `None`, than I'll get `settings.DEFAULT_FEATURE_ACTIVATION_THRESHOLD`

Reply:
I don't see any value to add a default value for a feature. I feel that features are so important that we should specify all its settings without default values. We can have a recommended value as a comment in the `Feature` class but they should be explicit put in the config files.

Reply:
In my understanding this is more a per network rule, like: all features in the mainnet should have at least 90% of miners support, but we could set a different number when we want some specific feature to behave differently, on a case by case basis.

I see there are three levels of flexibility for this, from the least flexible to the most:

1. The `threshold` is a constant per network, that is, every feature has the same `threshold`. In Bitcoin, it's like this. They have 90% for the mainnet and 75% for the testnet. That is, it's not possible to activate any feature in the mainnet if it's not supported by 90%+ of the miners.
2. The `threshold` is variable per feature, but there's a default for each network. That's what's currently implemented. It allows for flexibility, but gives preference to using the default value for the network.
3. Lastly, your suggestion, the `threshold` is variable per feature, and there's no default. Every time a new feature is configured, the threshold must be set.

I don't see a problem going either way, actually. The only point I think we have to have in mind is that every time a new feature is activated, it'll create a fork in the network, so the higher the `threshold` is, the more we are sure that most miners are not going to be forked away.

In other words, I set a default value for each network to provide a high default threshold, to minimize misuse of the feature activation process: although possible, maybe we shouldn't have lots of features getting activated with low thresholds.

That being said, our context is very different than Bitcoin's, as we are close to our partners and community, so it's very unlikely that our features are going to be rejected. Feature thresholds are still going to be discussed and reviewed every time we deploy a new feature.

So, I don't see a problem having a default, and that's what I would vote for. But if you think we should explicitly decide on a threshold for each feature, I also think that's a valid argument. I just think we should try to policy ourselves on not releasing many features with low thresholds (actually, we should do this anyway, even if there's a default).

Reply:
Got it. You convinced me to keep it as you did. Thanks for the clarification.